draft-cwbgp-ivy-energy-saving-management.md

```diff
@@ -186,20 +195,30 @@ informative:
    node as shown in Figure 2.
 
 ~~~~
-     augment /ni:network-elements/ni:network-element/ni:components/ni:component:
-       +--ro energy-consumption
-       |  +--ro total-energy-consumption?   uint64 <--- udpate the module
-       |  +--ro saved-energy                unit64
-       |  +--ro average-power?              gauge64 <--- udpate the module
-       |  +--ro current-power?              gauge64
-       |  +--ro saved-power?                int64
-       |  +--ro rated-power?                int64
-       |  +--ro instant-voltage?            int64
-       |  +--ro instant-current?            int64 <--- what this about?
-       |  +--ro temperature?                int32
-       +--rw energy-saving
-          +--rw enabled?      boolean
-          +--ro oper-state?   energy-saving-oper-state
+module: ietf-energy-saving-mgt
+  +--rw component-energy-monitoring
+     +--ro energy-consumption
+     |  +--ro average-power?     yang:gauge64
+     |  +--ro saved-power?       yang:gauge64
+     |  +--ro current-power?     yang:gauge64
+     |  +--ro current-volts?     int32
+     |  +--ro current-amperes?   int32
+     |  +--ro temperature?       int32
+     +--rw energy-saving
+     |  +--rw enabled?      boolean
+     |  +--ro oper-state?   identityref
+     +--ro inventory-component-ref?   -> /ni:network-elements/network-element/components/component/name
+
+  augment /ni:network-elements/ni:network-element/ni:components/ni:component:
```

Comment:
Why do we need those to be "rw" in the invenrory?

Reply:
yes, all of those can be read only, especially dynamic data, but energy saving mode, maybe we should still allow "rw", allow the clients control the energy daving mode they can select.

Reply:
So the node level is also readonly, like this:
  augment /nw:networks/nw:network/nw:node:
    +--ro energy-power-consumption
    |  +--ro total-energy-consumption?   yang:gauge64
    |  +--ro saved-energy?               yang:gauge64
    |  +--ro eer?                        decimal64
    +--ro energy-saving-modes
       +--ro energy-saving-mode* [mode]
          +--ro mode       identityref
          +--ro methods
             +--ro method* [method-name]
                +--ro method-name    identityref

Reply:
Since energy-saving-modes have been used in two different place, one is set as rw, the other place is set as ro, I am wondering whether we can define one grouping, reuses in two places, it looks hard. I am a little bit struggling about this.
Alternatively we don't define grouping, Just define energy saving modes container  in two places, but it seems clumsy.

Reply:
Yes, we can. Just remove config false from the grouping and refine it when called as appropriate

Reply:
can you give an example on how to refine?

Reply:
Here is an example from the AC spec for example: 

           uses bgp-svc {
             refine "peer-groups/peer-group/local-address" {
               config false;
             }
             refine "neighbor/local-address" {
               config false;
             }
           }